types/nexelec/origin/uplink.js

```diff
@@ -224,11 +224,8 @@ function handleHistoricalData(payload, productType, timestamp) {
 
   const periodMs = datalog.samplingPeriod * 60 * 1000;
 
-  // If timestamp is not provided, use current time
-  if (!timestamp) {
-    timestamp = Date.now();
-  }
-  const baseTimeMs = timestamp;
+  // Backwards compatibility
+  const baseTimeMs = (isNaN(timestamp) ? timestamp = new Date(timestamp) : timestamp = new Date(1000 * timestamp)).getTime();
```

Comment:
note that until now it was seconds and not milliseconds, please check the logic below